### ISSUE TITLE
Filter countries by region or specific clusters

### ISSUE THREAD

**Issue by pzwsk** [GFDRR/open-risk-data-dashboard#244]
In https://index.opendri.org/countries.html add a filter to show only countries belonging to the group selected.

To be discussed: should we handle this both BE and FE side, with in the db a specific column for group (one country can have belong to more than one group) or one group per column with a boolean for each country.

What do you think @nastasi-oq ?

Here is the first list of clusters and regions:
- Africa
- East Asia and Pacific
- Europe and Central Asia
- Latin America and Carribean
- Middle East and North Africa
- South Asia
- European Union
- Middle Income Countries
- Organization of Eastern Caribbean States
- Pacific Islands
- Small States
- Gulf Cooperation Council
- Western Europe
- Small Island Developing States

**Comment by nastasi-oq** [comment]
Specific columns for groups don't scale very well.
In my opinion we can use a ***many-to-many relation*** between the current **`Country`** table and a new **`Cluster`** table.
Another interesting option could be a ***many-to-many relation*** between items of the same table (a generalization of the current **`Country`** table) adding a **boolean field** to distinguish between a **cluster item** and a **country item**.
We need to evaluate pro and cons of both and then decide.

**Comment by nastasi-oq** [comment]
> Would Country_Group_Name + list of WB_Code be ok?
LGTM
A note: if you also want to use (or reuse a previous defined) coding schema for these groups you can also add it (like an arbitrary  3 char code to use a key).

**Comment by pzwsk** [comment]
Thanks, 

as per your suggestion, I found that WB is actually using 2 characters (apha or digit) codes for some of the country groups (not for all). They even have a code for World which is `1W`.

I suggest we use a "extended" version with our own codes for country groups that do not belong to this list.

So, please find below the updated list of countries (one change for World WB code), as well as list of Country Group. 

Let me know if ok

Country_List.csv
https://www.dropbox.com/s/4p2modfuv4euzro/Country_List.csv?dl=0

Country_Group.csv (Names of Country Group + WB Codes)
https://www.dropbox.com/s/ns8c7ygwc7yum7n/Country_Group.csv?dl=0

Country_Group_Details (Name of Country Group + list of countries WB code)
https://www.dropbox.com/s/1ef5nob61b5nigo/Country_Group_Details.csv?dl=0
https://www.dropbox.com/s/1ef5nob61b5nigo/Country_Group_Details.csv?dl=0

**Comment by pzwsk** [comment]
Also, once updated, could you put those csv files in the back end folder and remove lists that are unused? Thanks!

**Comment by pzwsk** [comment]
As discussed, moved to FE iteration #4

**Comment by nastasi-oq** [comment]
List of country groups with their associated countries from the CSV file, just for another visual check:

**Pacific Islands**
  FJ: Fiji
  FM: Micronesia, Federated States of
  KI: Kiribati
  MH: Marshall Islands
  PW: Palau
  SB: Solomon Islands
  TO: Tonga
  VU: Vanuatu
  WS: Samoa

**Small Island Developing States**
  AG: Antigua and Barbuda
  BS: Bahamas, The
  BH: Bahrain
  BB: Barbados
  BZ: Belize
  CV: Cabo Verde
  KM: Comoros
  CU: Cuba
  DM: Dominica
  DO: Dominican Republic
  FJ: Fiji
  GD: Grenada
  GW: Guinea-Bissau
  GY: Guyana
  HT: Haiti
  JM: Jamaica
  KI: Kiribati
  MV: Maldives
  MH: Marshall Islands
  MU: Mauritius
  FM: Micronesia, Federated States of
  NR: Nauru
  PW: Palau
  PG: Papua New Guinea
  WS: Samoa
  ST: Sao Tome and Principe
  SC: Seychelles
  SG: Singapore
  SB: Solomon Islands
  KN: St. Kitts and Nevis
  LC: St. Lucia
  VC: St. Vincent and the Grenadines
  SR: Suriname
  TP: Timor-Leste
  TO: Tonga
  TT: Trinidad and Tobago
  TV: Tuvalu
  VU: Vanuatu

**East Asia and Pacific**
  CN: China
  FJ: Fiji
  FM: Micronesia, Federated States of
  ID: Indonesia
  KH: Cambodia
  KI: Kiribati
  KR: Korea, Republic of
  LA: Lao People's Democratic Republic
  MH: Marshall Islands
  MM: Myanmar
  MN: Mongolia
  MY: Malaysia
  PG: Papua New Guinea
  PH: Philippines
  PW: Palau
  SB: Solomon Islands
  TH: Thailand
  TO: Tonga
  VN: Vietnam
  VU: Vanuatu
  WS: Samoa

**Europe and Central Asia**
  AL: Albania
  AM: Armenia
  AZ: Azerbaijan
  BA: Bosnia and Herzegovina
  BG: Bulgaria
  BY: Belarus
  CY: Cyprus
  CZ: Czech Republic
  EE: Estonia
  GE: Georgia
  HR: Croatia
  HU: Hungary
  KG: Kyrgyz Republic
  KZ: Kazakhstan
  LT: Lithuania
  LV: Latvia
  MD: Moldova
  MK: Macedonia, former Yugoslav Republic of
  PL: Poland
  PT: Portugal
  RO: Romania
  RU: Russian Federation
  SI: Slovenia
  SK: Slovak Republic
  TJ: Tajikistan
  TM: Turkmenistan
  TR: Turkey
  UA: Ukraine
  UZ: Uzbekistan
  YF: Serbia
**Latin America and The Carribean**
  AG: Antigua and Barbuda
  AR: Argentina
  BB: Barbados
  BO: Bolivia
  BR: Brazil
  BS: Bahamas, The
  BZ: Belize
  CL: Chile
  CO: Colombia
  CR: Costa Rica
  DM: Dominica
  DO: Dominican Republic
  EC: Ecuador
  GD: Grenada
  GT: Guatemala
  GY: Guyana
  HN: Honduras
  HT: Haiti
  JM: Jamaica
  KN: St. Kitts and Nevis
  LC: St. Lucia
  MX: Mexico
  NI: Nicaragua
  PA: Panama
  PE: Peru
  PY: Paraguay
  SR: Suriname
  SV: El Salvador
  TT: Trinidad and Tobago
  UY: Uruguay
  VC: St. Vincent and the Grenadines
  VE: Venezuela
**Middle East and North Africa**
  AE: United Arab Emirates
  BH: Bahrain
  DZ: Algeria
  EG: Egypt, Arab Republic of
  GZ: West Bank and Gaza
  IQ: Iraq
  IR: Iran, Islamic Republic of
  JO: Jordan
  KW: Kuwait
  LB: Lebanon
  LY: Libya
  MA: Morocco
  MT: Malta
  OM: Oman
  QA: Qatar
  RY: Yemen, Republic of
  SA: Saudi Arabia
  SY: Syrian Arab Republic
  TN: Tunisia
**South Asia**
  AF: Afghanistan
  BD: Bangladesh
  BT: Bhutan
  IN: India
  LK: Sri Lanka
  MV: Maldives
  NP: Nepal
  PK: Pakistan
**Sub-Saharan Africa**
  AO: Angola
  BF: Burkina Faso
  BI: Burundi
  BJ: Benin
  BW: Botswana
  CF: Central African Republic
  CG: Congo, Republic of
  CI: Cote d'Ivoire
  CM: Cameroon
  CV: Cabo Verde
  DJ: Djibouti
  ER: Eritrea
  ET: Ethiopia
  GA: Gabon
  GH: Ghana
  GM: Gambia, The
  GN: Guinea
  GQ: Equatorial Guinea
  GW: Guinea-Bissau
  KE: Kenya
  KM: Comoros
  LR: Liberia
  LS: Lesotho
  MG: Madagascar
  ML: Mali
  MR: Mauritania
  MU: Mauritius
  MW: Malawi
  MZ: Mozambique
  NA: Namibia
  NE: Niger
  NG: Nigeria
  RW: Rwanda
  SC: Seychelles
  SD: Sudan
  SL: Sierra Leone
  SN: Senegal
  SO: Somalia
  SS: South Sudan
  ST: Sao Tome and Principe
  SZ: Eswatini
  TD: Chad
  TG: Togo
  TZ: Tanzania
  UG: Uganda
  ZA: South Africa
  ZM: Zambia
  ZR: Congo, Democratic Republic of
  ZW: Zimbabwe
**Europe**
  AD: Andorra
  AL: Albania
  AT: Austria
  BA: Bosnia and Herzegovina
  BE: Belgium
  BG: Bulgaria
  BY: Belarus
  CH: Switzerland
  CZ: Czech Republic
  DE: Germany
  DK: Denmark
  EE: Estonia
  ES: Spain
  FI: Finland
  FR: France
  GB: United Kingdom
  GG: Guernsey, C.I.
  GI: Gibraltar
  GR: Greece
  HR: Croatia
  HU: Hungary
  IE: Ireland
  IM: Isle of Man
  IS: Iceland
  IT: Italy
  JE: Jersey, C.I.
  LI: Liechtenstein
  LT: Lithuania
  LU: Luxembourg
  LV: Latvia
  MC: Monaco
  MD: Moldova
  MK: Macedonia, former Yugoslav Republic of
  MT: Malta
  NL: Netherlands
  NO: Norway
  PL: Poland
  PT: Portugal
  RO: Romania
  RU: Russian Federation
  SE: Sweden
  SI: Slovenia
  SK: Slovak Republic
  TR: Turkey
  UA: Ukraine
  YF: Serbia

**Comment by pzwsk** [comment]
Yes I confirm, thank

**Comment by nastasi-oq** [comment]
New countries list
-------
 ``http://www.example.com/api/country/``

![country](https://user-images.githubusercontent.com/1670278/48630814-740e8700-e9bd-11e8-9c31-662b0922a679.png)

How to retrieve country_groups list
-------
 ``http://www.example.com/api/country_group/``

![country_group](https://user-images.githubusercontent.com/1670278/48630836-7e308580-e9bd-11e8-8993-26b91829bfb1.png)

**Comment by oncletom** [comment]
@nastasi-oq small question: is there a way to know in which group(s) a country is from the `/api/country/` endpoint?

So far, my understanding is the API has to be queried to get to know this answer.

**Comment by nastasi-oq** [comment]
> @nastasi-oq small question: is there a way to know in which group(s) a country is from the /api/country/ endpoint?

No, I explicitly asked about this option and the answer is that it isn't an exposed information.

**Comment by nastasi-oq** [comment]
@pzwsk 

> one minor thing: API response for country code says iso2 but this not iso
> code but wb_ code or wb_id Could we change that? This may create confusion
> in the future.

I agree and thinking about this not backward compatible change I also decided to investigate the possibility to move the new ``wb_id`` as primary key of Country model, making more readable any relation in the database, working on it since Monday I'm analyzing result of a refactored migration and I plan to open a new PR that MUST be merged before Country Group PR.

**Comment by pzwsk** [comment]
Ok sounds good, thanks

**Comment by nastasi-oq** [comment]
Implemented in #328
